src/components/Button/Button.tsx

```diff
@@ -0,0 +1,7 @@
+import * as S from './style'
+import { ButtonProps } from './Button.types'
+
+
+export const Button = ({label, onClick}:ButtonProps) => {
+    return <S.Button onClick={onClick}>{label}</S.Button>
```

Comment:
Com as mudanças que propus na tipagem, você terá que utilizar todas as props default do botão e repassar para o componente. Dessa forma:

```suggestion
export const Button = ({label, ...props}:ButtonProps) => {
    return <S.Button {...props}>{label}</S.Button>
```

Reply:
Entendi! Eu e o Mov conversamos na hora essa possibilidade e decidimos optar por uma forma mais simples e clara pra explicar pra eles. Vale depois a gente precisa trocar uma ideia e chegar em um concenso sobre aprofundamos mais (com isso aumenta a complexidade e abaixa a compreenção do conhecimento) pra aumentar a qualidade.